algorithms/merge-sort.js

```diff
@@ -1 +1,33 @@
+function mergeSort(arr) {
+  let lHalf = arr.slice(0, arr.length / 2);
+  let rHalf = arr.slice(arr.length / 2, arr.length);
+  sort(lHalf);
+  sort(rHalf);
+  arr = merge(lHalf, rHalf);
+  return arr;
+}
 
+function sort(arr) {
+  let temp;
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr.length; j++) {
+      if (arr[i] < arr[j]) {
+        temp = arr[i];
+        arr[i] = arr[j];
+        arr[j] = temp;
+      }
+    }
+  }
+  return arr;
+}
+
+function merge(lHalf, rHalf) {
+  let mergedArray = [...lHalf, ...rHalf];
+  sort(mergedArray);
```

Comment:
This also plays a part and removes the advantage merge sort gives us. Combining two sorted arrays into one array just creates an unsorted array again. Which means you call the `sort()` func which has a runtime of `n^2`